public/app/core/components/AppChrome/MegaMenu/MegaMenu.tsx

```diff
@@ -64,6 +64,10 @@ export const MegaMenu = memo(
     const onPinItem = (id?: string) => {
       if (id && config.featureToggles.pinNavItems) {
         const newItems = isPinned(id) ? pinnedItems.filter((i) => id !== i) : [...pinnedItems, id];
+        const interactionName = isPinned(id) ? 'grafana_nav_item_unpinned' : 'grafana_nav_item_pinned';
+        reportInteraction(interactionName, {
+          itemId: id,
```

Comment:
In the `grafana_navigation_item_clicked` event we send `path: newItem.url ?? newItem.id,` instead of `itemId`, maybe we should do the same so its easier to compare?